cmd/git_utils.go

```diff
@@ -129,19 +137,55 @@ func GetGitInfo(config *RootCommandConfig) (GitInfo, error) {
 		}
 
 	}
+
 	if strings.Contains(value, noCommits) {
 		gitInfo.Branch = StringAfter(value, noCommits)
 	}
+
 	changesMade := false
-	outputLength := len(outputLines)
+	outputLength := len(statusOutputLines)
 
 	if outputLength > 1 {
 		changesMade = true
-
 	}
 	gitInfo.ChangesMade = changesMade
 
-	errMsg := ""
+	if gitInfo.Upstream == "" {
+		config.Debug.log("Failed to retrieve upstream from git status -sb. Looking via latest commit")
+		outputLines, err := RunGitBranchContains(config.LoggingConfig, gitInfo.Commit.SHA, config.ProjectDir, lineSeparator, config.Dryrun)
+		if err != nil {
+			errMsg += "Unable to locate latest commit in remote. "
+		} else {
+			gitInfo.Upstream = outputLines[0]
+			for _, upstream := range outputLines {
+				if upstream == "origin" {
+					gitInfo.Upstream = upstream
+					break
+				} else if upstream == "upstream" {
+					gitInfo.Upstream = upstream
+				}
+			}
+			gitInfo.Upstream = strings.TrimSpace(gitInfo.Upstream)
+		}
+	}
+
+	if gitInfo.Upstream == "" {
+		gitInfo.Commit.Pushed = false
+		gitRemoteOutput, err := RunGitRemote(config.LoggingConfig, config.ProjectDir, lineSeparator, config.Dryrun)
```

Comment:
I would also add a check in `RunGitRemote` to add an error if it's empty, not very common but could happen if the user did a `git init` and didn't add a remote manually.